### PR TITLE
TSPS-224 add admin endpoint to be able to update workspace id for a pipeline

### PR DESCRIPTION
**Pull request by jsotobroad**
### Description 

In order to make e2e testing easier (and for generally more flexibility in running tsps) we want to add an admin endpoint that allows an admin user to update the workspace id for a given pipeline.

### Jira Ticket
https://broadworkbench.atlassian.net/browse/TSPS-224

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,6 +55,8 @@ To run locally:
 4. Run `scripts/write-config.sh`
 5. Run `./gradlew bootRun` to spin up the server.
 6. Navigate to [http://localhost:8080/#](http://localhost:8080/#)
+7. If this is your first time deploying to any environment, be sure to use the use the admin endpoint `/api/admin/v1/updatePipelineWorkspaceId/{pipelineName}/{workspaceId}` to set your pipeline's workspace id
```
Comment (mmorgantaylor) [inline README.md:58]: nit: "use the" is duplicated

also: would be good to point in the right direction for where to find the workspaceId to use

File: common/openapi.yml [hunk 2]
```diff
@@ -3,40 +3,30 @@ info:
...
-          $ref: '#/components/responses/ServerError'
-        503:
-          $ref: '#/components/responses/SystemStatusResponse'
-  /version:
+  # Admin APIs
+  /api/admin/v1/updatePipelineWorkspaceId/{pipelineName}/{workspaceId}:
+    parameters:
+      - $ref: '#/components/parameters/PipelineName'
+      - $ref: '#/components/parameters/WorkspaceId'
     get:
```
Comment (mmorgantaylor) [inline common/openapi.yml:11]: should this be a GET? seems like PUT might be more appropriate?

Comment (mmorgantaylor) [reply common/openapi.yml:11]: (was wondering if we do want a GET method here to just retrieve what workspaceId the service is pointing to)

File: service/src/main/java/bio/terra/pipelines/app/controller/PipelineApiUtils.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package bio.terra.pipelines.app.controller;
+
+import bio.terra.pipelines.common.utils.PipelinesEnum;
+import bio.terra.pipelines.db.exception.InvalidPipelineException;
+import org.slf4j.Logger;
+
+public class PipelineApiUtils {
```
Comment (mmorgantaylor) [inline service/src/main/java/bio/terra/pipelines/app/controller/PipelineApiUtils.java:7]: nice, this is great

File: service/src/test/java/bio/terra/pipelines/controller/AdminApiControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  private final String testUserId = testUser.getSubjectId();
+
+  @BeforeEach
+  void beforeEach() {
+    when(samUserFactoryMock.from(any(HttpServletRequest.class), any())).thenReturn(testUser);
+    doNothing().when(samServiceMock).checkAdminAuthz(testUser);
+  }
+
+  @Test
+  void updatePipeineWorkspaceIdOk() throws Exception {
```
Comment (mmorgantaylor) [inline service/src/test/java/bio/terra/pipelines/controller/AdminApiControllerTest.java:55]: nit: typo "Pipeine"

File: service/src/test/java/bio/terra/pipelines/service/PipelinesServiceTest.java [hunk 5]
```diff
@@ -86,8 +91,25 @@ void testPipelineHashCode() {
...
           p.hashCode());
     }
   }
+
+  @Test
+  void updatePipelineWorkspaceId() {
+    PipelinesEnum pipelinesEnum = PipelinesEnum.IMPUTATION_MINIMAC4;
+    Pipeline p = pipelinesService.getPipeline(pipelinesEnum);
+    UUID savedWorkspaceId = UUID.randomUUID();
+    // make sure the current pipeline does not have the workedspace id we're trying to update with
```
Comment (mmorgantaylor) [inline service/src/test/java/bio/terra/pipelines/service/PipelinesServiceTest.java:105]: nit: typo "workedspace"

**Review by mmorgantaylor — APPROVED** [review]
looks good, some minor nits, and one question about the API verb

File: service/src/test/java/bio/terra/pipelines/service/PipelinesServiceTest.java (hunk 5 above)
Comment (jsotobroad) [reply service/src/test/java/bio/terra/pipelines/service/PipelinesServiceTest.java:105]: i swear i can spell :(

File: README.md [hunk 6]
```diff
@@ -55,6 +55,8 @@ To run locally:
 4. Run `scripts/write-config.sh`
 5. Run `./gradlew bootRun` to spin up the server.
 6. Navigate to [http://localhost:8080/#](http://localhost:8080/#)
+7. If this is your first time deploying to any environment, be sure to use the admin endpoint `/api/admin/v1/updatePipelineWorkspaceId/{pipelineName}/{workspaceId}` to set your pipeline's workspace id. Workspace id can be found through the terra ui workspace dashboard or through the Rawls GET workspace endpoint.
```
Comment (mmorgantaylor) [inline README.md:58]: workspace id actually doesn't show up (in an easily accessible way) on the terra ui workspace dashboard :(

File: common/openapi.yml (hunk 2 above)
Comment (jsotobroad) [reply common/openapi.yml:10]: yea thats a great idea, added a GET method and cleaned up the apis a little

File: README.md (hunk 6 above)
Comment (jsotobroad) [reply README.md:58]: oh i usually just use the "bucket" name copy thing under "cloud information" to find the workspace id.  its usually like `sc-{workspace_uuid}` or `fc-{workspace_uuid}`.  I guess it depends on your definition of easily accessible hahaha

Comment (mmorgantaylor) [reply README.md:58]: fair - I get annoyed when I have to copy something into another text editor and edit it further. but you're right that it shows up there

**Pull request merged**